ompi/mca/osc/rdma/osc_rdma_component.c

```diff
@@ -372,26 +372,37 @@ static int ompi_osc_rdma_component_query (struct ompi_win_t *win, void **base, s
                                           struct ompi_communicator_t *comm, struct opal_info_t *info,
                                           int flavor)
 {
+
+    opal_list_t *btls = NULL;
+    mca_btl_base_selected_module_t* selected_btl;
+
     if (MPI_WIN_FLAVOR_SHARED == flavor) {
         return -1;
     }
 
-    /* GPU buffers are not supported by the rdma component */
-    if (MPI_WIN_FLAVOR_CREATE == flavor) {
-        uint64_t flags;
-        int dev_id;
-        if (opal_accelerator.check_addr(*base, &dev_id, &flags)) {
-            return -1;
-        }
-    }
-
     /* verify if we have any btls available.  Since we do not verify
      * connectivity across all btls in the alternate case, this is as
      * good a test as we are going to have for success. */
     if (opal_list_is_empty(&mca_btl_base_modules_initialized)) {
         return -1;
     }
 
+    btls = &mca_btl_base_modules_initialized;
+
+    OPAL_LIST_FOREACH(selected_btl, btls, mca_btl_base_selected_module_t) {
+        opal_output_verbose(MCA_BASE_VERBOSE_INFO, ompi_osc_base_framework.framework_output,
+                    "osc_rdma_component_query: check ACCELERATOR_RDMA flag: %s",
+                    selected_btl->btl_component->btl_version.mca_component_name);
+        mca_btl_base_module_t *btl = selected_btl->btl_module;
+        // Check flag: MCA_BTL_FLAGS_ACCELERATOR_RDMA
+        if (btl->btl_flags & MCA_BTL_FLAGS_ACCELERATOR_RDMA) {
+            goto end_check;
+        }
+    }
+    return -1;
```

Comment:
This isn't exactly correct. We still want to work if this returns false - 
```
      if (opal_accelerator.check_addr(*base, &dev_id, &flags)) {
```
We still want to be able to use the OSC RDMA component when no gpu btl's are available if we're not given a GPU BTL.

Reply:
I get your points, so we need double-check, if it is on GPU, then if any of them can have RDMA, we let it pass.

I will fix it.